util/train_test_functions.R

```diff
@@ -411,3 +486,88 @@ PredictWrapper <- function(train.model.list, pred.list, sample.df,
   }
 
 }
+
+mean_one_versus_all_AUC <- function(probabilities_matrix,
+                                    true_subtypes) {
+  
+  # Calculate the one vs. all AUC values for each subtype and return the mean
+  #
+  # Args:
+  #   probabilities_matrix: matrix with one column per subtype, one row per
+  #     sample, and with each row summing to 1, usually the output of predict()
+  #   true_subtypes: vector of true subtype labels
+  #
+  # Returns:
+  #   AUC value (mean of one vs. all AUCs)
+  
+  true_subtypes <- factor(true_subtypes)
+  
+  if (!all(sort(levels(true_subtypes)) == sort(colnames(probabilities_matrix)))) {
```

Comment:
Do these need to be in the same order or no? Which is a way of asking is this check that has `sort()` in it going to be sufficient?

Reply:
Maybe this is handled in `PredictAUC()`?

Reply:
`levels(true_subtypes)` and `colnames(probabilities_matrix)` do not need to be in the same order for actually running `mean_one_versus_all_AUC()`. The purpose of this check was to ensure the subtype levels match the column names one for one /or that the set of subtypes equals the set of column names.

I can see moving this check into `PredictAUC()`, and also reducing the number of calls to `mean_one_versus_all_AUC()` by moving it outside the `if else` structure.